packages/expo-payments-stripe/plugin/src/withStripe.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  AndroidConfig,
+  ConfigPlugin,
+  createRunOncePlugin,
+  IOSConfig,
+  withAndroidManifest,
+  withEntitlementsPlist,
+  withXcodeProject,
+} from '@expo/config-plugins';
+
+const {
+  prefixAndroidKeys,
+  getMainApplicationOrThrow,
+  addMetaDataItemToMainApplication,
+  removeMetaDataItemFromMainApplication,
+} = AndroidConfig.Manifest;
+
+const pkg = require('expo-payments-stripe/package.json');
+
+type StripePluginProps = {
+  scheme: string;
+  /**
+   * The iOS merchant ID used for enabling Apple Pay.
+   * Without this, the error "Missing merchant identifier" will be thrown on iOS.
+   */
+  merchantId: string;
+};
+
+const CUSTOM_TAB_ACTIVITY = 'expo.modules.payments.stripe.RedirectUriReceiver';
+const META_WALLET = 'com.google.android.gms.wallet.api.enabled';
+
+function buildXMLItem({
+  head,
+  children,
+}: {
+  head: Record<string, string>;
+  children?: Record<string, string | any[]>;
+}) {
+  return { ...(children || {}), $: head };
+}
+
+function buildAndroidItem(name: string | Record<string, any>) {
+  const item = typeof name === 'string' ? { name } : name;
+  const head = prefixAndroidKeys(item);
+  return buildXMLItem({ head });
+}
+
+// From Android.Facebook
+function getStripeSchemeActivity(scheme: string): AndroidConfig.Manifest.ManifestActivity {
+  /**
+     <activity
+        android:name="expo.modules.payments.stripe.RedirectUriReceiver"
+        ...
+        android:value="true">
+        <intent-filter>
+            <action android:name="android.intent.action.VIEW" />
+            <category android:name="android.intent.category.DEFAULT" />
+            <category android:name="android.intent.category.BROWSABLE" />
+            <data android:scheme="${scheme}" />
+        </intent-filter>
+    </activity>
+       */
+
+  return buildXMLItem({
+    head: prefixAndroidKeys({
+      name: CUSTOM_TAB_ACTIVITY,
+      exported: 'true',
+      theme: '@android:style/Theme.Translucent.NoTitleBar.Fullscreen',
+      launchMode: 'singleTask',
+    }),
+    children: {
+      'intent-filter': [
+        {
+          action: [buildAndroidItem('android.intent.action.VIEW')],
+          category: [
+            buildAndroidItem('android.intent.category.DEFAULT'),
+            buildAndroidItem('android.intent.category.BROWSABLE'),
+          ],
+          data: [buildAndroidItem({ scheme })],
+        },
+      ],
+    },
+  }) as AndroidConfig.Manifest.ManifestActivity;
+}
+
+export function ensureStripeActivity({
+  mainApplication,
+  scheme,
+}: {
+  mainApplication: AndroidConfig.Manifest.ManifestApplication;
+  scheme: string;
+}): AndroidConfig.Manifest.ManifestApplication {
+  if (Array.isArray(mainApplication.activity)) {
+    // Remove all Facebook CustomTabActivities first
+    mainApplication.activity = mainApplication.activity.filter(activity => {
+      return (activity.$ || {})['android:name'] !== CUSTOM_TAB_ACTIVITY;
```

Comment:
What does the $ mean?

Reply:
In the XML package we use across the ecosystem `$` represents the head keys. 
Something like this: `<activity name="foo"></activity>` would become `{ $: { name: 'foo' } }`